### ISSUE TITLE
Use a Connection Pool in Hastory-Cli

### ISSUE THREAD

**Issue by StevenXL** [NorfairKing/hastory#35]
Please see #33 for conversation, but to summarize the issue:

Currently, any command that utilize the database runs migrations on every request to the db before making the request. Instead, we should be setting up the database once, and then using a connection pool.

**Comment by StevenXL** [comment]
@NorfairKing Hi! This issue was pulled out from a discussion that we had on the code review for PR #33. (I have taken screenshots of the relevant parts of that PR for your easy reference; see below). 

I am questioning whether this PR is valid. The reason I ask is that the `hastory` project does not have a daemon running. This means that whenever we run a `hastory` command, we will have to `initialize` the application all over again. If this is the case, does it matter if we initialize the DB in `combineToInstructions`, or if it is initialized when the command uses the DB. In the first case, we are always initializing the DB, regardless of whether it is used or not by the command. In the second case, we'd save ourselves the trouble of initializing the DB if it is not used. 

Currently, usage of the DB looks something like this:

| Command                | DB Requirement |
|------------------------|----------------|
| gather                 | Always         |
| genGatherWrapperScript | Never          |
| change                 | Sometimes      |
| listRecentDirs         | Sometimes      |
| genChangeWrapperScript | Never          |
| suggest                | Always         |
| sync                   | Always         |
|                        |                |
|                        |                |

One way that I think we can improve things is to `not` run migrations if the database already exist, though I do not know the implications of this if we update the project and someone downloads the updated `hastory`. I don't know if my concerns are well-founded or not, so maybe a call to go over this might be appropriate? Let me know. 

<img width="803" alt="Screen Shot 2020-10-25 at 9 02 09 AM" src="https://user-images.githubusercontent.com/7391433/97109428-2585bf00-16a1-11eb-983b-e0c0eb0d9f50.png">

<img width="798" alt="Screen Shot 2020-10-25 at 9 01 03 AM" src="https://user-images.githubusercontent.com/7391433/97109436-29194600-16a1-11eb-98b1-0f46ed9ade6e.png">

**Comment by NorfairKing** [comment]
Hello hello

It's good that you pushed back, because it is indeed wrong to put the `ConnectionPool` in the settings.
The reason for that is that you need `withSqlitePool` (or so), so you can't build it in the `combineToInstructions`. We'd have to have a separate `Env` that the commands can use, that this pool would be in.

However, the main reason I suggested this, is that it's the standard way to do things.
It _may_ be the case that there's no point for this project, but then we should describe that in depth, which is more work than to just do things the standard way.

To answer some of your other questions
1. We _must_ run the migrations on every run, because the data format may have changed and the migration takes care of resolving that.
2. Your db usage table raises a good point: We should not be creating and migrating a database if no database access is needed. This may complicate things. Especially the `genXYZScript` commands need to be as fast as possible because they run on shell boot.


TL;DR You're probably right, but we should add a comment to the `runDB` that this function should only be called once per command invocation.

**Comment by StevenXL** [comment]
Closed due to #42.
